update-v0.c

```diff
@@ -0,0 +1,261 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <error.h>
+#include <errno.h>
+#include <string.h>
+#include <stdint.h>
+#include <sys/types.h>
+
+#include "micro.h"
+#include "crc8.h"
+#include "update-v0.h"
+
+#define TS7970_I2C_ADDR 0x10
+
+/* Read-back status values */
+/* Default value of status, closed */
+#define STATUS_CLOSED 0x00
+/* Once the flashwrite process is set up, but no data written */
+#define STATUS_READY 0xAA
+/* Flashwrite process has seen full length of data written and is considered done */
+#define STATUS_DONE 0x01
+/* Flashwrite is in process, meaning SOME data has been written, but not the full length */
+#define STATUS_IN_PROC 0x02
+/* A CRC error occurred at ANY point during data write. Note that this status
+ * is not set if CRC fails for open process, the system simply does not open
+ */
+#define STATUS_CRC_ERR 0x03
+/* An error occurred while trying to erase the actual flash */
+#define STATUS_ERASE_ERR 0x04
+/* An error occurred at ANY point during data write. */
+#define STATUS_WRITE_ERR 0x05
+/* Erase was successful, but, the area to be written was not blank */
+#define STATUS_NOT_BLANK 0x06
+/* A BSP error opening and closing flash. Most errors are buggy code, configurations, or unrecoverable */
+#define STATUS_OPEN_ERR 0x07
+/* Wait state while processing a write */
+#define STATUS_WAIT 0x08
+/* Request the uC reboot at any time after its open status */
+#define STATUS_RESET 0x55
```

Comment:
These shouldn't be repeated if common between v0 and v1. Can stay as #defines in a common update header, or enums, etc.

Reply:
Fixed